app/javascript/controllers/count_controller.js

```diff
@@ -0,0 +1,25 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static targets = ["counter", "display"]
+  static values = {
+    selected: { type: Number, default: 0 }
+  }
+
+  connect() {
+    this.selectedValue = this.countSelectedTargets()
+    this.displayTarget.innerHTML = this.selectedValue
+  }
+
+  recount() {
+    this.selectedValue = this.countSelectedTargets()
+    this.displayTarget.innerHTML = this.selectedValue
+  }
+
+  countSelectedTargets() {
```

Comment:
i wasn't quite sure how to generalize this action, so for now it is reading the `disabled` property on the inputs within the dates. and then calculating which dates are selected by determining which have been enabled.

Reply:
It is a little awkward to not just count the `counterTargets`.

Could you instead add this to the `form.hidden_field` in `_calendar.html.erb`?  Would basically look the same, except getting rid of the `firstElementChild`.  Maybe a bit cleaner?

Or it's nearly the same - your call.